### DISCUSSION TITLE
Shouldn't review results show as events

### DISCUSSION THREAD

**Discussion by matheusfillipe** [pwntester/octo.nvim#358] [Ideas]
It is common to have multiple reviews from different persons over the time with different status from each one like `changes requested` or `approved`. 

Is it currently possible to show those in the events list. I find it just not enough to have this in the summary. Would this be a desired feature?

**Comment by pwntester** [comment]
Thanks for the feedback. These are the current events that are shown in Octo https://github.com/pwntester/octo.nvim/blob/24a171868df3dc9eba61cda393b3c5baa6896201/lua/octo/model/octo-buffer.lua#L177-L205

```lua
    elseif item.__typename == "AssignedEvent" then
      writers.write_assigned_event(self.bufnr, item)
      prev_is_event = true
    elseif item.__typename == "PullRequestCommit" then
      writers.write_commit_event(self.bufnr, item)
      prev_is_event = true
    elseif item.__typename == "MergedEvent" then
      writers.write_merged_event(self.bufnr, item)
      prev_is_event = true
    elseif item.__typename == "ClosedEvent" then
      writers.write_closed_event(self.bufnr, item)
      prev_is_event = true
    elseif item.__typename == "ReopenedEvent" then
      writers.write_reopened_event(self.bufnr, item)
      prev_is_event = true
    elseif item.__typename == "LabeledEvent" then
      table.insert(unrendered_labeled_events, item)
    elseif item.__typename == "UnlabeledEvent" then
      table.insert(unrendered_unlabeled_events, item)
    elseif item.__typename == "ReviewRequestedEvent" then
      writers.write_review_requested_event(self.bufnr, item)
      prev_is_event = true
    elseif item.__typename == "ReviewRequestRemovedEvent" then
      writers.write_review_request_removed_event(self.bufnr, item)
      prev_is_event = true
    elseif item.__typename == "ReviewDismissedEvent" then
      writers.write_review_dismissed_event(self.bufnr, item)
      prev_is_event = true
    end
```

There are plenty of events though:
- https://docs.github.com/en/graphql/reference/unions#pullrequesttimelineitems
- https://docs.github.com/en/graphql/reference/unions#issuetimelineitems

Showing new events should be very simple, just add new `elseif` block for the type you would like to be shown and add a new writer for it (you can see the others for examples).

Would you be willing to send a PR for this?

**Reply by matheusfillipe** [reply]
It seems that change requests is already implemented. Just the approvals that aren't shown in the event list.  I am not sure from which event that comes from. Maybe [PullRequestReview ](https://docs.github.com/en/graphql/reference/objects#pullrequestreview)?

**Reply by matheusfillipe** [reply]
Oh actually you can approve a pull request without reviewing it. That is the thing I am missing

**Comment by williambdean** [comment]
Hi @matheusfillipe 

I've created an issue for this here: https://github.com/pwntester/octo.nvim/issues/951
